### PR TITLE
[test] Unify fake events system. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: test/browser/fake_events.js [hunk 1]
```diff
@@ -2,51 +2,57 @@
...
 
-function simulateKeyUp(keyCode, code = undefined) {
-  simulateKeyEvent('keyup', keyCode, code);
+function simulateKeyUp(target, keyCode, code = undefined) {
+  simulateKeyEvent(target, 'keyup', keyCode, code);
 }
 
-function simulateMouseEvent(x, y, button, absolute) {
+function simulateKeyDownUp(target, keyCode, code = undefined) {
+  simulateKeyEvent(target, 'keyodwn', keyCode, code);
```
Comment (kripken) [inline test/browser/fake_events.js:26]:
```suggestion
  simulateKeyEvent(target, 'keydown', keyCode, code);
```

File: test/browser/fake_events.js [hunk 2]
```diff
@@ -2,51 +2,57 @@
...
+  simulateKeyEvent(target, 'keyup', keyCode, code);
```
Comment (kripken) [inline test/browser/fake_events.js:27]: Should `simulateKeyDownUp` perhaps call `simulateKeyDown, simulateKeyUp`? That would reuse more code and avoid hardcoding the event names here.

Comment (sbc100) [reply test/browser/fake_events.js:27]: Oops, yes of course.

File: test/browser/fake_events.js (hunk 1 above)
Comment (sbc100) [reply test/browser/fake_events.js:26]: Oops.

File: test/browser/test_sdl2_mouse.c [hunk 3]
```diff
@@ -90,11 +90,11 @@ int main() {
   int absolute = false;
 #endif
 
-  EM_ASM(simulateMouseEvent(10, 20, -1, $0), absolute); // move from 0,0 to 10,20
-  EM_ASM(simulateMouseEvent(10, 20, 0, $0), absolute); // click
-  EM_ASM(simulateMouseEvent(10, 20, 0, $0), absolute); // click some more, but this one should be ignored through PeepEvent
-  EM_ASM(simulateMouseEvent(30, 70, -1, $0), absolute); // move some more
-  EM_ASM(simulateMouseEvent(30, 70, 1, $0), absolute); // trigger the end
+  EM_ASM(simulateMouseMove(10, 20, $0), absolute); // move from 0,0 to 10,20
```
Comment (kripken) [inline test/browser/test_sdl2_mouse.c:93]: This has one too few parameters, I think? New code is
```js
function simulateMouseMove(x, y, button, absolute) {
```

Comment (sbc100) [reply test/browser/test_sdl2_mouse.c:93]: Ooops, fixed.  simulateMouseMove should only take 3 args.

**Comment by sbc100** [comment]
I refactored this change so that most existing callsites do not need to change.

**Review by kripken — APPROVED** [review]
lgtm but I would check with all relevant `interactive` tests before landing.

**Comment by sbc100** [comment]
> lgtm but I would check with all relevant `interactive` tests before landing.

Actually, I looked through the interactive tests and none of them actually use this stuff, since interactive tests tend to rely on real events not fake ones.

**Pull request merged**